backend/compact-connect/stacks/persistent_stack/staff_users.py

```diff
@@ -68,13 +68,29 @@ def __init__(
         if not callback_urls:
             raise ValueError(
                 "This app requires a callback url for its authentication path. Either provide 'domain_name' or set "
-                "'allow_local_ui' to true in this environment's context.",
+                "'allow_local_ui' to true in this environment's context."
             )
 
+        logout_urls = []
+        if stack.ui_domain_name is not None:
+            logout_urls.append(f'https://{stack.ui_domain_name}/Login')
+            logout_urls.append(f'https://{stack.ui_domain_name}/Logout')
+        # This toggle will allow front-end devs to point their local UI at this environment's user pool to support
+        # authenticated actions.
+        if environment_context.get('allow_local_ui', False):
+            local_ui_port = environment_context.get('local_ui_port', '3018')
+            logout_urls.append(f'http://localhost:{local_ui_port}/Login')
+            logout_urls.append(f'http://localhost:{local_ui_port}/Logout')
+        if not logout_urls:
+            raise ValueError(
+                "This app requires a logout url for its logout function. Either provide 'domain_name' or set "
+                "'allow_local_ui' to true in this environment's context."
+            )
```

Comment:
So Im on the fence about whether to alter to the comment.

I feel that there are potentially pros and cons,  the cons might be me misinterpreting things, let me know. Id imaging I would want to move the callback URL code just above it as well, if I do?

Pros:
Remove repeated code.

Cons:
UserPool construct becomes not a general class construct but more opinionated into how this project is specifically configured. Because these are common_constructs just for this project in this case that may not be a real con.

Reply:
I see your point with the con. Perhaps there is another place we could put the common code for generating this project specific logout urls then? I'll leave it up to your call. If you're good with how it is. You can resolve this comment.

Reply:
I think I want to leave it for now because it will be easier to change if we want the user pools to not function identically in the future. 

That being said, there is repeated code / logic not just between user pools but between callback urls and logout urls that might be better as a general thing because it seems like we will often generally run into wanting to allow domain name redirection and localhost redirection.

Ill make a very low priority tech ticket to do refactor that + and revisit the repeated code with callbackUrls + logOutUrls by that time if they still havent changed it might make sense to cement / abstract that away.

Does that sound good to you?

Reply:
Yep that sounds reasonable. Thanks.

Reply:
The user pool construct is intended to be specialized/opinionated, so I'd be in favor of consolidating.

Reply:
Well it appears Im outvoted. Ill make the callback url and logout url consolidation change now, but not the url generation function I mentioned